client/chat/route.tsx

```diff
@@ -1,30 +1,56 @@
-import React from 'react'
-import { useRoute } from 'wouter'
-import { makeSbChannelId } from '../../common/chat'
+import React, { Suspense } from 'react'
+import { Route, RouteProps, Switch } from 'wouter'
+import { makeSbChannelId, SbChannelId } from '../../common/chat'
+import { hasAnyPermission } from '../admin/admin-permissions'
 import { replace } from '../navigation/routing'
+import { LoadingDotsArea } from '../progress/dots'
+import { useAppSelector } from '../redux-hooks'
 import { ConnectedChatChannel } from './channel'
+import { ChannelList } from './channel-list'
+import { CreateChannel } from './create-channel'
 
-export function ChannelRouteComponent(props: { params: any }) {
-  const [matches, params] = useRoute('/chat/:channelId/:channelName')
+const LoadableChatAdmin = React.lazy(async () => ({
+  default: (await import('./admin')).ChatAdmin,
+}))
+
+export function ChannelRoute({
+  component: Component,
+  ...rest
+}: Omit<RouteProps, 'component'> & {
+  component: React.ComponentType<{ channelId: SbChannelId; channelName: string }>
+}) {
+  return (
+    <Route<{ channelId: string; channelName: string }> {...rest}>
+      {params => {
+        const channelIdNum = Number(params.channelId)
+        if (isNaN(channelIdNum)) {
+          queueMicrotask(() => {
+            replace('/')
+          })
+          return null
+        }
 
-  if (!matches) {
-    queueMicrotask(() => {
-      replace('/')
-    })
-    return null
-  }
-  const channelIdNum = Number(params.channelId)
-  if (isNaN(channelIdNum)) {
-    queueMicrotask(() => {
-      replace('/')
-    })
-    return null
-  }
+        return (
+          <Component channelId={makeSbChannelId(channelIdNum)} channelName={params.channelName} />
+        )
+      }}
+    </Route>
+  )
+}
+
+export function ChannelRouteComponent(props: { params: any }) {
+  // TODO(2Pac): Add a separate permission for managing chat channels
```

Comment:
Why? What purpose would that serve?

Reply:
Hmm, my thinking was that we might have channel moderators that shouldn't have access to change channel's banners and descriptions. But maybe that's all a part of channel moderation?

Reply:
I think you are confusing "channel moderation" as in "people that have mod in a channel" and "channel moderation" as in "people who can access and moderate every channel on the server". The latter seems like such a wide permission that I don't see why we would trust people to do it but couldn't also trust that same person to manage a banner setting or something